backend/pkg/schema/client.go

```diff
@@ -687,35 +688,45 @@ func (c *Client) CreateSchema(ctx context.Context, subjectName string, schema Sc
 
 // GetSchemasIndividually returns all schemas by describing all schemas one by one. This may be used against
 // schema registry that don't support the /schemas endpoint that returns a list of all registered schemas.
-func (c *Client) GetSchemasIndividually(ctx context.Context, showSoftDeleted bool) ([]SchemaVersionedResponse, error) {
+func (c *Client) GetSchemasIndividually(ctx context.Context, showSoftDeleted bool) ([]*SchemaVersionedResponse, []error) {
 	subjectsRes, err := c.GetSubjects(ctx, showSoftDeleted)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get subjects to fetch schemas for: %w", err)
+		return nil, []error{fmt.Errorf("failed to get subjects to fetch schemas for: %w", err)}
 	}
 
 	g, ctx := errgroup.WithContext(ctx)
 	g.SetLimit(10) // limit max concurrency to 10 requests at a time
 
-	schemas := make([]SchemaVersionedResponse, 0, len(subjectsRes.Subjects))
+	schemas := make([]*SchemaVersionedResponse, 0, len(subjectsRes.Subjects))
+	errors := make([]error, len(subjectsRes.Subjects))
+	hasErrors := atomic.Bool{}
```

Comment:
I think we should do `errors := make([]error, 0, len(subjectsRes.Subjects))` and then we can check the length of errors to see if we have errors. I think in most cases we do not expect every single subject to return an error.

It feels a bit weird to return a slice of schemas and a slice of errors. At this point we can probably just refactor the function to return schema + error via a channel and handle the error and returned schema on the receiving side.

Reply:
The intent is to be able to get all the schemas we successfully fetched, even if we failed to retrieve some subset of all the schemas in SR. So we return all the successfully fetched schemas and a collection of errors.
This is the case for both `GetSchemasIndividually()` and `GetSchemas()`, and then `GetProtoDescriptors()` we are able to log all the failed schemas, but still be able to parse and compile the successfully retrieved ones so we could potentially use them.